src/main/kotlin/ink/abb/pogo/scraper/tasks/EvolvePokemon.kt

```diff
@@ -42,49 +47,56 @@ class EvolvePokemon : Task {
+                        countDown.countDown()
+                        val result = it.response
+                        Log.yellow("Result of using lucky egg: ${result.result.toString()}")
+                    }
+                    countDown.await()
                 } catch (exc: Exception) {
                     Log.red("Lucky egg usage failed! Will continue evolving stack without one.")
                 }
             } else {
                 Log.yellow("Starting stack evolve of $countEvolveStack pokemon without lucky egg")
             }
             var countEvolved = 0
-            ctx.api.inventories.pokebank.pokemons.forEach {
-                if (settings.evolveBeforeTransfer.contains(it.pokemonId)) {
-                    val pokemonMeta = PokemonMetaRegistry.getMeta(it.pokemonId)
-                    if (bot.api.inventories.candyjar.getCandies(pokemonMeta.family) >= pokemonMeta.candyToEvolve) {
-                        Log.yellow("Evolving ${it.pokemonId.name} CP ${it.cp} IV ${it.getIvPercentage()}%")
-                        val evolveResult = it.evolve()
-                        if( settings.evolveTimeDelay > 300){
-                            Thread.sleep(settings.evolveTimeDelay/2 + (Math.random()*settings.evolveTimeDelay).toLong())
-                        } else {
-                            Thread.sleep(300)
-                        }
-                        if (evolveResult.isSuccessful) {
-                            countEvolved++
-                            val evolvedpokemon = evolveResult.evolvedPokemon
-                            Log.yellow("Successfully evolved in ${evolvedpokemon.pokemonId.name} CP ${evolvedpokemon.cp} IV ${evolvedpokemon.getIvPercentage()}%")
-                            ctx.server.releasePokemon(it.id)
-                            //TODO: comunicate to the sockserver the new got pokemon
-                            //ctx.server.newPokemon(0.0, 0.0, PokemonDataOuterClass.PokemonData.buildFromPokemon(evolvedpokemon))
-                            ctx.api.inventories.updateInventories(true)
-                            Thread.sleep(300)
-                        } else {
-                            Log.red("Evolve of ${it.pokemonId.name} CP ${it.cp} IV ${it.getIvPercentage()}% failed: ${evolveResult.result.toString()}")
+            ctx.api.inventory.pokemon.forEach {
+                if (settings.evolveBeforeTransfer.contains(it.value.pokemonData.pokemonId)) {
+                    val pokemonMeta = PokemonMetaRegistry.getMeta(it.value.pokemonData.pokemonId)
+                    if (bot.api.inventory.candies.getOrPut(pokemonMeta.family, { AtomicInteger(0) }).get() >= pokemonMeta.candyToEvolve) {
+                        val pokemonData = it.value.pokemonData
+                        Log.yellow("Evolving ${pokemonData.pokemonId.name} CP ${pokemonData.cp} IV ${pokemonData.getIvPercentage()}%")
+                        val evolve = EvolvePokemon().withPokemonId(it.key)
```

Comment:
Looks like there's still a minor candy cache problem:

```
26 aug 12:03:45 [default: ProfileLoop] - Stack of pokemon ready to evolve: 1/1
26 aug 12:03:45 [default: ProfileLoop] - Starting stack evolve of 1 pokemon without lucky egg
26 aug 12:03:45 [default: ProfileLoop] - Evolving RATTATA CP 10 IV 84%
26 aug 12:03:46 [default: ProfileLoop] - Successfully evolved in RATICATE CP 19 IV 84%
26 aug 12:03:46 [default: ProfileLoop] - Not enough candy (7/12) to evolve WEEDLE CP 28 IV 73%
26 aug 12:03:46 [default: ProfileLoop] - Not enough candy (7/12) to evolve PIDGEY CP 10 IV 42%
26 aug 12:03:46 [default: ProfileLoop] - Evolving RATTATA CP 10 IV 46%
26 aug 12:03:47 [default: ProfileLoop] - Evolve of RATTATA CP 10 IV 46% failed: FAILED_INSUFFICIENT_RESOURCES
26 aug 12:03:47 [default: ProfileLoop] - Finished evolving 1 pokemon; 0 xp gained
```

On the second rattata he still thinks there is enough candy, even in the next loop he does:

```
26 aug 12:04:45 [default: ProfileLoop] - Stack of pokemon ready to evolve: 1/1
26 aug 12:04:45 [default: ProfileLoop] - Starting stack evolve of 1 pokemon without lucky egg
26 aug 12:04:45 [default: ProfileLoop] - Not enough candy (7/12) to evolve WEEDLE CP 28 IV 73%
26 aug 12:04:45 [default: ProfileLoop] - Not enough candy (7/12) to evolve PIDGEY CP 10 IV 42%
26 aug 12:04:45 [default: ProfileLoop] - Evolving RATTATA CP 10 IV 46%
26 aug 12:04:46 [default: ProfileLoop] - Evolve of RATTATA CP 10 IV 46% failed: FAILED_INSUFFICIENT_RESOURCES
26 aug 12:04:46 [default: ProfileLoop] - Finished evolving 0 pokemon; 1050 xp gained
```